includes/class-bitcoin-donation-shortcode-voting.php

```diff
@@ -0,0 +1,228 @@
+        $theme_class = $dark_mode == true ? "dark-theme" : "light-theme";
+        $active = get_post_meta($poll_id, '_bitcoin_donation_polls_active', true);
+        $one_vote = get_post_meta($poll_id, '_bitcoin_donation_polls_one_vote', true);
+
+        // Check if poll is active
+        if (!$active) {
+            ob_start();
+?>
+            <div id="bitcoin-voting-form" class="bitcoin-voting-form  <?php echo esc_attr($theme_class); ?>">
+                <div class="bitcoin-voting-form-container">
+                    <h3><?php echo esc_html($title ? $title : 'Bitcoin Voting'); ?></h3>
+                    <p><?php echo esc_html($description ? $description : 'What would you like to see more of on our blog?'); ?></p>
+                    <h4>This poll is not active</h4>
+                </div>
+            </div>
+        <?php
+            return ob_get_clean();
+        }
+
+        // Fetch options from meta data
+        for ($i = 1; $i <= 4; $i++) {
+            $option = get_post_meta($poll_id, "_bitcoin_donation_polls_option_{$i}", true);
+            if (!empty($option)) {
+                $options[$i] = $option;
+                $num_options++;
+            }
+        }
+
+        $options2 = get_option('bitcoin_donation_forms_options');
+        $options2 = is_array($options2) ? $options2 : [];
+        ob_start();
+
+        $now = current_time('timestamp');
+        $start_timestamp = strtotime($start_date);
+        $end_timestamp = strtotime($end_date);
+        error_log($now);
+        error_log($start_timestamp);
+        error_log($end_timestamp);
+        if ($now < $start_timestamp) {
+            $time_until_start = human_time_diff($now, $start_timestamp);
+        ?>
+            <div id="bitcoin-voting-form" class="bitcoin-voting-form  <?php echo esc_attr($theme_class); ?>">
+                <div class="bitcoin-voting-form-container">
+                    <h3><?php echo esc_html($title ? $title : 'Bitcoin Voting'); ?></h3>
+                    <p><?php echo esc_html($description ? $description : 'What would you like to see more of on our blog?'); ?></p>
+                    <h4>Poll starting in: <?php echo $time_until_start; ?></h4>
+                </div>
+            </div>
+        <?php
+        } elseif ($now > $end_timestamp) {
+        ?>
+            <div id="bitcoin-voting-form" class="bitcoin-voting-form  <?php echo esc_attr($theme_class); ?>">
+                <div class="bitcoin-voting-form-container">
+                    <h3><?php echo esc_html($title ? $title : 'Voting Poll'); ?></h3>
+                    <p><?php echo esc_html($description ? $description : ''); ?></p>
+                    <div id="poll-results" class="poll-results" data-end-date="<?php echo $end_date; ?>" data-poll-id="<?php echo $poll_id; ?>">
+                        <?php
+                        for ($i = 1; $i <= min(4, $num_options ? $num_options : 4); $i++):
```

Comment:
This can be improved. There are a couple more places in this file where the same variable is redundant.
```suggestion
                        for ($i = 1; $i <= min(4, $num_options ?: 4); $i++):
```